modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4ChunkedHttpResponse.java

```diff
@@ -38,4 +39,14 @@ public ChunkedRestResponseBody body() {
     public int getSequence() {
         return sequence;
     }
+
+    @Override
+    public void addHeader(String name, String value) {
+        headers().add(name, value);
+    }
+
+    @Override
+    public boolean containsHeader(String name) {
+        return headers().contains(name);
+    }
```

Comment:
Personally I feel it is natural for `Netty4HttpResponse` to extend `HttpResponse` so that these methods can be kept at interface level. Or we should have a comment somewhere to explain why future refactor should not make such change.

Reply:
I'm currently working on a (possible) enhancement that would allow us to split responses over multiple such messages, for which it only makes sense for the first message in the sequence to extend `HttpResponse` and carry stuff like headers. If that lands then this'll explain why we've cut down this interface, and if it doesn't then we can pull these back up to the super-interface.